### PR TITLE
#158457038 Start A New Game And Save Game Players

### PR DESCRIPTION
**Pull request by kenware**
#### What does this PR do?
- Start a new game
- Save the user that started the game, players to the database.
#### Description of Task to be completed?
- Design a modal that prompt users to start a game
- Start a game
- Create an authenticated route/middleware
- Create controllers and models for the authenticated route
- Create angular functions that make HTTP response to save game players
#### How should this be manually tested?
- Clone this repository and install dependency
- Npm start
- Click on the start button (A modal will prompt you to start a game)
- On game session after there are at least three players. Start button will start the game and save the 
  game starter, players, and gameID to the database
#### What are the relevant pivotal tracker stories?
Users should be able to create/start a new game

#### Screenshots

screen shot 2018-07-12 at 10 02 05 am
screen shot 2018-07-12 at 10 01 28 am

### CODE REVIEW

**Review by adetola-ralph — CHANGES_REQUESTED** [review]
> - There are files in this PR that you were not meant to touch please fix
> - Redesign the modal so that it is consistent with the theme of the game
> - How do you have only one commit for everything that you worked on


-The second point in the feedback; It is done.
-The third point, Our team agreed to make the number of a commit as minimal as possible for easy review.
Therefore, I mostly used " git commit --amend --no-edit" command to avoid many commit message

-For the first point in the feedback, the babel-env that is messing with .babelrc configuration when packages are installed is still in the develop branch. I will uninstall it and push.

**Comment by kenware** [comment]
The first point, i need more clarity
-Second point, It is done but i have not pushed till I have more clarity.
-Third point, Our team agreed to make number of commit as minimal as possible for easy review.
Furthermore, I only have one commit because I and blessing pulled from dika's branch to work on the UI before it was merged to develop branch. Blessing's commit was lost after rebasing off develop branch, I decided to transfer all my changes  to a new branch  which is up to date from develop branch. Basically we shouldn't  pull from another branch and rebase on develop branch to avoid our commit being lost.

File: app/controllers/game.js
Comment (adetola-ralph) [inline app/controllers/game.js]: Fix the typo here

File: app/controllers/game.js
Comment (adetola-ralph) [inline app/controllers/game.js]: 401 isn't the right status code to be sent back when an error occurs during an operation on the server

File: app/controllers/game.js
Comment (adetola-ralph) [reply app/controllers/game.js]: I don't think `gameStarter` is necessary here. All that is needed is the `gameId` & `players`. What value will saving the `gameStarter` bring to the application

File: app/middleware/auth.js
Comment (adetola-ralph) [inline app/middleware/auth.js:4]: This comment is placed wrongly, it is not above the comment for which it is providing commentary for

File: app/middleware/auth.js
Comment (adetola-ralph) [inline app/middleware/auth.js:26]: Space after this line

File: app/middleware/auth.js
Comment (adetola-ralph) [inline app/middleware/auth.js]: There is no comment for this function, also what is the difference between this function and the `verifyToken` function below it

File: app/middleware/auth.js
Comment (adetola-ralph) [inline app/middleware/auth.js]: Space after this line

File: app/models/game.js
Comment (adetola-ralph) [inline app/models/game.js]: As I mentioned above, what is the value added by having this field here

File: backend-test/startGame/startGame.js
Comment (adetola-ralph) [inline backend-test/startGame/startGame.js]: The comment states that `POST /api/auth/endpoint should return the user token along with the`, along with the what?

File: backend-test/startGame/startGame.js
Comment (adetola-ralph) [inline backend-test/startGame/startGame.js]: Space after this block

File: backend-test/startGame/startGame.js
Comment (adetola-ralph) [inline backend-test/startGame/startGame.js]: Space after this block

File: backend-test/startGame/startGame.js
Comment (adetola-ralph) [inline backend-test/startGame/startGame.js]: try catch block here is unnecessary

File: backend-test/startGame/startGame.js
Comment (adetola-ralph) [inline backend-test/startGame/startGame.js]: Remove this line

File: backend-test/startGame/startGame.js
Comment (adetola-ralph) [inline backend-test/startGame/startGame.js]: Once again this try catch block is unnecessary

File: config/routes.js
Comment (adetola-ralph) [inline config/routes.js]: The way `verifyToken` is used here means that it should be in a controller file and not a middleware file. By the way I don't think verifyToken is necessary, you can have a middleware that check that all the protected routes are protected and sends a 401 back for a bad token so the frontend can redirect the user to the login screen

Comment (adetola-ralph) [reply config/routes.js]: This is redundant here, the get route can be chained to the api instance we have on line 14. So it would look like 
```
api
  .post('bla bla')
  .get('yor route here')
```

File: config/routes.js
Comment (adetola-ralph) [inline config/routes.js]: Same issue as my 2nd comment directly above this

File: config/socket/game.js
Comment (adetola-ralph) [inline config/socket/game.js]: this should be `userId` not `userID`

File: config/socket/socket.js
Comment (adetola-ralph) [inline config/socket/socket.js]: `userId` not `userID`

File: public/js/controllers/game.js
Comment (adetola-ralph) [inline public/js/controllers/game.js]: Remove friends, it is not necessary

File: public/js/controllers/game.js
Comment (adetola-ralph) [inline public/js/controllers/game.js]: This block of code is not necessary either

**Review by adetola-ralph — CHANGES_REQUESTED** [review]
> - Why are you testing user authentication in start game?
I had to login to obtain a token before testing the game route as it requires a token(authenticated route
)

File: app/middleware/auth.js
Comment (kenware) [reply app/middleware/auth.js:4]: Done

File: app/controllers/game.js
Comment (kenware) [reply app/controllers/game.js]: Done

File: app/middleware/auth.js
Comment (kenware) [reply app/middleware/auth.js:26]: Done

File: app/middleware/auth.js
Comment (kenware) [reply app/middleware/auth.js]: -Commented
-Removed verify token function

File: config/routes.js
Comment (kenware) [reply config/routes.js]: Done

File: public/js/controllers/game.js
Comment (kenware) [reply public/js/controllers/game.js]: Done

File: public/js/controllers/game.js
Comment (kenware) [reply public/js/controllers/game.js]: Done

File: app/controllers/game.js
Comment (kenware) [reply app/controllers/game.js]: Noted

**Review by adetola-ralph — APPROVED** [review]

**Pull request merged**
